testing/camunda-process-test-spring/src/main/java/io/camunda/process/test/impl/configuration/CamundaProcessTestRuntimeConfiguration.java

```diff
@@ -47,8 +51,15 @@ public class CamundaProcessTestRuntimeConfiguration {
 
   private CamundaProcessTestRuntimeMode runtimeMode = CamundaProcessTestRuntimeMode.MANAGED;
 
+  private boolean preferLegacyConfiguration = false;
+
   @NestedConfigurationProperty private RemoteConfiguration remote = new RemoteConfiguration();
 
+  @PostConstruct
+  public void init() {
+    this.preferLegacyConfiguration = legacyConfiguration.isCustomConfiguration();
+  }
```

Comment:
:wrench: Instead of checking the legacy configuration in the getters, we could set the fields during the construction. 

```
@Configuration
@EnableConfigurationProperties(LegacyCamundaProcessTestRuntimeConfiguration.class)
@ConfigurationProperties(prefix = "camunda.process-test")
public class CamundaProcessTestRuntimeConfiguration {

  public CamundaProcessTestRuntimeConfiguration(
      final LegacyCamundaProcessTestRuntimeConfiguration legacyConfiguration) {
      // override the fields by the legacy configuration before setting the properties  
      BeanUtils.copyProperties(legacyConfiguration, this);
  }

}
```

I found this pattern in the Camunda codebase (e.g., `OperatePropertiesOverride`).